src/ansiblelint/rules/AnsibleSyntaxCheckRule.py

```diff
@@ -39,7 +41,15 @@ def _get_ansible_syntax_check_matches(lintable: Lintable) -> List[MatchError]:
             return []
 
         with timed_info("Executing syntax check on %s", lintable.path):
-            cmd = ['ansible-playbook', '--syntax-check', str(lintable.path)]
+            extra_vars_cmd = []
+            if options.extra_vars:
+                extra_vars_cmd = ['--extra-vars', json.dumps(options.extra_vars)]
```

Comment:
This will likely fail to work well for more complex vars, like one containing shell escapable vars, multilines and so.

IMHO, I would instead dump these vars inside `.cache/extra_vars.yml` and pass `@.cache/extra_vars.yml` to ansible, so it will load them from there.

Be sure to add some problematic var example to the list and see if it works. I really do not want to get bug reports about linter failing to load var files, when this is the job of ansible itself.

Maybe we should only support mentioning the extra_vars_filename inside config and force users that want this feature to add another file with them, so we can rely on ansible to process the file.

Reply:
I don't see why this should fail for more complex variables. The shell is not involved in interpreting this (see `shell=False` below), this is passed directly as-is to ansible-playbook.

Reply:
I will be glad to see it as working as is, lets put something like this into the example, which also works as test-file:
```
extra_vars:
  # sample values used for testing
  foo: |
    line1
    line2
  bar: ":{;\t$()"
```